From 46943a73e5569ebb99e764196d2a6ec59086afe2 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Tue, 16 Dec 2014 22:22:10 -0500
Subject: [scsi] cxgb4i: fix credit check for tx_data_wr

Message-id: <1418768535-2637-3-git-send-email-svemuri@redhat.com>
Patchwork-id: 102107
O-Subject: [RHEL7.1 PATCH BZ 1174982 2/7] cxgb4i: fix credit check for tx_data_wr
Bugzilla: 1174982
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

Upstream : https://git.kernel.org/cgit/linux/kernel/git/davem/net-next.git/commit/?id=7857c62a35041a21a66ccab551601c942b748330

make sure any tx credit related checking is done before adding the wr header.

Signed-off-by: Karen Xie <kxie@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 7857c62a35041a21a66ccab551601c942b748330)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

Conflicts:
 drivers/scsi/cxgbi/cxgb4i/cxgb4i.c

diff --git a/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c b/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c
index f119a67..abee611 100644
--- a/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c
+++ b/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c
@@ -547,10 +547,11 @@ static inline void make_tx_data_wr(struct cxgbi_sock *csk, struct sk_buff *skb,
  struct fw_ofld_tx_data_wr *req;
  unsigned int submode = cxgbi_skcb_ulp_mode(skb) & 3;
  unsigned int wr_ulp_mode = 0;
+ bool imm = is_ofld_imm(skb);
 
  req = (struct fw_ofld_tx_data_wr *)__skb_push(skb, sizeof(*req));
 
- if (is_ofld_imm(skb)) {
+ if (imm) {
   req->op_to_immdlen = htonl(FW_WR_OP(FW_OFLD_TX_DATA_WR) |
      FW_WR_COMPL(1) |
      FW_WR_IMMDLEN(dlen));
-- 
1.7.1